pandas/tests/util/test_assert_series_equal.py

```diff
@@ -194,3 +195,39 @@ def test_series_equal_categorical_mismatch(check_categorical):
             tm.assert_series_equal(s1, s2, check_categorical=check_categorical)
     else:
         _assert_series_equal_both(s1, s2, check_categorical=check_categorical)
+
+
+@pytest.mark.parametrize("check_dtype", [True, False])
+def test_assert_series_equal_extension_dtype_mismatch(check_dtype):
+    left = Series(pd.array([1, 2, 3], dtype="Int64"))
+    right = left.astype(int)
+
+    msg = """Attributes of Series are different
+
+Attribute "dtype" are different
+\\[left\\]:  Int64
+\\[right\\]: int64"""
+
+    if check_dtype:
+        with pytest.raises(AssertionError, match=msg):
+            tm.assert_series_equal(left, right, check_dtype=check_dtype)
+    else:
+        tm.assert_series_equal(left, right, check_dtype=check_dtype)
```

Comment:
Can you just test both True / False cases inline here without the parametrization? (it actually gets you rid of the if/else, while for the rest not adding any code)

Reply:
(and same below)